src/main/SDL_main_callbacks.c

```diff
@@ -147,3 +147,13 @@ void SDL_QuitMainCallbacks(SDL_AppResult result)
     SDL_Quit();
 }
 
+void SDL_CheckDefaultArgcArgv(int *argc, char ***argv)
+{
+    if (!argv)
+    {
+        static char dummyargv0[] = { 'S', 'D', 'L', '_', 'a', 'p', 'p', '\0' };
+        static char *argvdummy[2] = { dummyargv0, NULL };
+        *argc = 1;
+        *argv = argvdummy;
+    }
+}
```

Comment:
The main reason I didn't declare the fallback argv as a global in #12676 is because the argv is mutable and programs are theoretically allowed to modify it. I don't know if SDL is supposed to support initializing and quitting SDL itself multiple times in the same process (like calling `SDL_RunApp()` multiple times from a test harness) but if it is then a regular local variable stored on the stack is the more sound alternative to globals.

Reply:
Like main(), this doesn't reset if you were to call into it again. We don't have any documentation to that effect for SDL_RunApp, but since these might literally be the values from a real main(), they should be assumed to be global memory.